gettsim/transfers/grundrente.py

```diff
@@ -0,0 +1,391 @@
+    cond = ~below_half_höchstwert & (
+        durchschnittl_entgeltpunkte_grundr < höchstwert_grundr_m
+    )
+    out.loc[cond] = (höchstwert_grundr_m - durchschnittl_entgeltpunkte_grundr) * (
+        1 - 0.125
+    )
+
+    # Case 3: Entgeltpunkte above Höchstwert
+    cond = durchschnittl_entgeltpunkte_grundr > höchstwert_grundr_m
+    out.loc[cond] = 0
+
+    # Set to 0 if Grundrentenzeiten below minimum
+    gr_zeiten_below_min = (
+        grundrentenzeiten < ges_renten_vers_params["grundrentenzeiten"]["min"]
+    )
+    out.loc[gr_zeiten_below_min] = 0
+
+    return out
+
+
+def zu_verst_eink_excl_grundr_zuschlag_m_tu(
+    zu_verst_eink_excl_grundr_zuschlag_m: FloatSeries, tu_id: IntSeries
+) -> FloatSeries:
+    """Aggregate the income relevant for income crediting rule of
+    Grundrentenzuschlag on tax unit level.
+
+    Parameters
+    ----------
+    zu_verst_eink_excl_grundr_zuschlag_m
+        See :func:`zu_verst_eink_excl_grundr_zuschlag_m`.
+    tu_id
+        See basic input variable :ref:`tu_id <tu_id>`.
+
+    Returns
+    -------
+
+    """
+    return zu_verst_eink_excl_grundr_zuschlag_m.groupby(tu_id).sum()
+
+
+def zu_verst_eink_excl_grundr_zuschlag_m(
+    proxy_rente_vorj_excl_grundr_zuschlag_m: FloatSeries,
+    bruttolohn_vorj_m: FloatSeries,
+    brutto_eink_1: FloatSeries,
+    brutto_eink_6: FloatSeries,
+    kapital_eink_minus_pauschbetr: FloatSeries,
+) -> FloatSeries:
+    """Income relevant for income crediting rule of Grundrentenzuschlag. The
+    Grundrentenzuschlag (in previous years) is not part of the relevant income and
+    does not lower the Grundrentenzuschlag.
+
+    The Deutsche Rentenversicherung uses the income of the year two to three years
+    ago to be able to use administrative data on this income for the calculation.
+
+    "Es ist davon auszugehen, dass dem Finanzamt regelmäßig zwei Jahre nach dem Ablauf
+    des Veranlagungszeitraums die Daten vorliegen, die von der Rentenversicherung
+    abgerufen werden können. "
+
+    Warning: Currently, earnings of dependend work and pensions are based on the
+    last year, and other income on the current year instead of the year
+    two years ago to avoid the need for several new input variables.
+
+    #ToDo: Freibeträge for income are currently not considered
+    #ToDO: as freibetraege_tu depends on pension income
+
+    Parameters
+    ----------
+    proxy_rente_vorj_excl_grundr_zuschlag_m
+        See :func:`proxy_rente_vorj_excl_grundr_zuschlag_m`.
+    bruttolohn_vorj_m
+        See :func:`bruttolohn_vorj_m`.
+    brutto_eink_1
+        See :func:`brutto_eink_1`.
+    brutto_eink_6
+        See :func:`brutto_eink_6`.
+    kapital_eink_minus_pauschbetr
+        See :func:`kapital_eink_minus_pauschbetr`.
+    Returns
+    -------
+    """
+
+    # Earnings from self-employed work and rent income (assumption that they were
+    # the same last year)
+    earnings_self_empl_rent = brutto_eink_1 + brutto_eink_6
```

Comment:
I think we should have a separate function for that.

Reply:
I decided to just use the two variables directly. I don't see the need for aggregating them.

Reply:
Okay!